### DISCUSSION TITLE
Question about BC

### DISCUSSION THREAD

**Discussion by caojinbo** [zwicker-group/py-pde#153] [Q&A] [answered]
Thanks for the great py-pde work.
I have a question about parameters in bc. for example, the document says
`
bc_lower = {'value': 4}
bc_upper = {'value': 2}
bc = [bc_lower, bc_upper]
`
What do 'lower' and 'upper' mean here? I tested these bc in Laplace equation, and found that x lower and upper is 4 (left and right side), the y lower and upper is 2 (top and bottom).

The other parameter seems to work as expected
`bc=[[bc_x_lower, bc_x_upper], [bc_y_lower, bc_y_upper]]`

**Comment by david-zwicker** [comment, marked as answer]
Boundaries are always perpendicular to axes in `py-pde`. The `lower` and `upper` then refers to the orientation of the axis. For example, in a 3d Cartesian system, the `lower` boundary condition along the `x`-direction is the yz-plane with the lowest value of the `x`-coordinate, while the `upper` boundary is the yz-plane with a large value of `x`. I guess it would help to have a schematic drawing in [the documentation](https://py-pde.readthedocs.io/en/latest/manual/advanced_usage.html#boundary-conditions), but I didn't find the time to include one, yet. If you could make a schematic drawing to explain the situation, I would be happy to include it!

**Comment by caojinbo** [comment]
Thank you. If you are OK, I do want to do something with the documentation. The module is very powerful and it derserves more visibility.

**Reply by david-zwicker** [reply]
Yes, definitely. Extending the documentation or adding examples are definitely welcomed! The best thing would be if you could clone the repository, extend the documentation, and create a pull request that I could include in the main branch.
